### PR TITLE
[2201.1.x] Fix completions after isolated kw in block node

### LINKED ISSUE #37039: Invalid completions given for expression statements
**Description:**
After the addressing changes for https://github.com/ballerina-platform/ballerina-lang/issues/36920, 
Ballerina gives invalid completions for following code with the changes suggested in https://github.com/ballerina-platform/ballerina-lang/pull/36973

```
import ballerina/module1;

public const int TEST_CONST = 12;

public function main() {
    isolated f      // invalid completions given in here
    int[] testArr = [];
}

public type TestMap2 map<string>;

public type TestMap3 map<int>;

type TestMap1 map<int>;

type TestRecord1 record {
    int rec3Field1 = 12;
    string rec3Field2 = "TestRecord3";
};
```

Currently, Test-case related to this issue is skipped in https://github.com/ballerina-platform/ballerina-lang/pull/36973.
Please refer to the following comment.
https://github.com/ballerina-platform/ballerina-lang/pull/36973#discussion_r923052426

**Related Issues (optional):**
https://github.com/ballerina-platform/ballerina-lang/issues/36920

**Suggested Labels (optional):**
Bug

### PR DESCRIPTION
**Pull request by IMS94**
## Purpose
$subject

Fixes #37039

Related PRs: #37406

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/BlockNodeContextProvider.java [hunk 1]
```diff
@@ -194,6 +197,24 @@ protected List<LSCompletionItem> getStatementCompletionItems(BallerinaCompletion
         return completionItems;
     }
 
+    @Override
+    protected List<LSCompletionItem> getCompletionItemsOnQualifiers(Node node, BallerinaCompletionContext context) {
+        List<LSCompletionItem> completionItems = new ArrayList<>(super.getCompletionItemsOnQualifiers(node, context));
+        if (node.kind() == SyntaxKind.MODULE_VAR_DECL) {
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/BlockNodeContextProvider.java:203]: Is this a possible scenario inside a block node context?

**Review by malinthar — COMMENTED** [review]
Changes look good. Just a minor concern

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/BlockNodeContextProvider.java (hunk 1 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/BlockNodeContextProvider.java:203]: Yes, with the new parser change it's possible. See the issue #37039 and parser PR: #36973. This happens only after `isolated f<cursor>`

Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/BlockNodeContextProvider.java:203]: Wait. Just got what you said. Seems like you are correct.

Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/BlockNodeContextProvider.java]: Done!

**Review by malinthar — APPROVED** [review]
LGTM!

**Pull request merged**